packages/contracts/src/contract_wrappers/generated/base_contract.ts

```diff
@@ -27,9 +61,10 @@ export class BaseContract {
         }
         return txDataWithDefaults;
     }
-    constructor(web3ContractInstance: Web3.ContractInstance, defaults?: Partial<TxData>) {
-        this.address = web3ContractInstance.address;
-        this._web3ContractInstance = web3ContractInstance;
-        this._defaults = defaults || {};
+    constructor(web3Wrapper: Web3Wrapper, abi: Web3.ContractAbi, address: string) {
+        this._web3Wrapper = web3Wrapper;
+        this.abi = abi;
+        this.address = address;
+        this._ethersInterface = new ethersContracts.Interface(abi);
```

Comment:
Hm... this file seems duplicated... Can we not have a single `baseContract` in `0x.js` and `contracts`?